src/stream/src/task/barrier_manager.rs

```diff
@@ -118,60 +110,85 @@ impl LocalBarrierWorker {
             barrier_senders: HashMap::new(),
             failure_actors: HashMap::default(),
             state: ManagedBarrierState::new(state_store, streaming_metrics),
-            collect_complete_receiver: HashMap::default(),
+            epoch_result_sender: HashMap::default(),
         }
     }
 
     async fn run(mut self, mut event_rx: UnboundedReceiver<LocalBarrierEvent>) {
-        while let Some(event) = event_rx.recv().await {
-            match event {
-                LocalBarrierEvent::RegisterSender { actor_id, sender } => {
-                    self.register_sender(actor_id, sender);
-                }
-                LocalBarrierEvent::InjectBarrier {
-                    barrier,
-                    actor_ids_to_send,
-                    actor_ids_to_collect,
-                    result_sender,
-                } => {
-                    let result =
-                        self.send_barrier(&barrier, actor_ids_to_send, actor_ids_to_collect);
-                    let _ = result_sender.send(result).inspect_err(|e| {
-                        warn!(err=?e, "fail to send inject barrier result");
-                    });
-                }
-                LocalBarrierEvent::Reset => {
-                    self.reset();
-                }
-                ReportActorCollected { actor_id, barrier } => self.collect(actor_id, &barrier),
-                ReportActorFailure { actor_id, err } => {
-                    self.notify_failure(actor_id, err);
-                }
-                LocalBarrierEvent::CollectEpoch {
-                    epoch,
-                    result_sender,
-                } => {
-                    let result = self.remove_collect_rx(epoch);
-                    let _ = result_sender.send(result).inspect_err(|e| {
-                        warn!(err=?e.as_ref().map(|_|()), "fail to send collect epoch result");
-                    });
+        loop {
+            let item = drop_either_future(
```

Comment:
Can you elaborate more on the use of `drop_either_future` here?

Why we just drop lhs / rhs future?

Reply:
Why don't we use `tokio::select!` instead?

Reply:
I've changed to `tokio::select!`. Original use of `drop_either_future` is because the future will hold a mutable reference to self, and if not dropped we cannot modify the states in self.